src/commands.ts

```diff
@@ -87,26 +87,33 @@ async function deployCurrentProject(args, commandObserver: CommandObserver) {
 		project = args.fsPath;
 	}
 
-	await projectHelper.buildProjects([project], commandObserver).then(async result => {
-		if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
-			var filePath = await projectHelper.getOutputFilePath(project);
-			if (filePath && fs.existsSync(filePath)) {
-				var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
-				if (connection) {
-					vscode.workspace.openTextDocument(filePath).then(doc => {
-						vscode.window.showTextDocument(doc, vscode.ViewColumn.Active, false).then(() => {
-							let filePath = doc.uri.toString();
-							azdata.queryeditor.connect(filePath, connection.connectionId).then(() => azdata.queryeditor.runQuery(filePath, undefined, false));
+	try {
+		await projectHelper.buildProjects([project], commandObserver).then(async result => {
+			if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
+				await projectHelper.setOutputFilePath(project, commandObserver);
+				var filePath = commandObserver.outputFilePath;
+				if (filePath && fs.existsSync(filePath)) {
+					var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
+					if (connection) {
+						let projectFileText = fs.readFileSync(filePath, 'utf8');
+						azdata.queryeditor.connect(filePath, connection.connectionId).then(() => {
+							commandObserver.logToOutputChannel(localize('extension.deploymentStartedMessage', '\nDeployment started {0}.', new Date().toLocaleString()));
+							commandObserver._client.sendRequest("query/executeDeployString", {owner_uri: filePath, query:projectFileText})
+							.then(() => { }, err => {
+								commandObserver.logToOutputChannel(localize('extension.FailedDeploy', 'Deployment failed with error: {0}', err.message));
```

Comment:
Would we get multiple errors? If it still breaks it down to different statements and executes one at a time, then we would get multiple errors, right?

Reply:
This error is only when something is wrong with the request. The current execute implementation is such that if there is an error it would stop executing any further queries.

Reply:
Ok, the execute sounds reasonable - if we fail to create a table, there is no reason to attempt to create objects that depend on it. Although if error is "table already exists", then we might want to just continue. We can do it later though, I don't think it's super critical and we need to better understand "update" scenario, when some objects already present.